src/tracker.rs

```diff
@@ -48,6 +63,20 @@ impl Tracker {
         }
     }
 
+    /// Replaces the tracker subject with the provided subject
+    ///
+    /// Requires a mutable reference
+    pub fn replace_subject(&mut self, subject: Subject) {
+        self.subject = subject;
+    }
+
+    /// Updates the tracker subject with the provided subjects values
+    ///
+    /// Requires a mutable reference
+    pub fn update_subject(&mut self, subject: Subject) {
```

Comment:
Nit: not sure what are the Rust conventions for setters, but I wonder if we need two setter functions? I have seen that in some Rust code, classes provide access to a mutable reference of their properties – check out the `first_name_mut()` example here: https://stackoverflow.com/a/35391084

Reply:
Ooo, I like that a lot more than my updating methods, good idea.